### ISSUE TITLE
Invites can now return a error code of M_UNSUPPORTED_ROOM_VERSION

### ISSUE THREAD

**Issue by erikjohnston** [vector-im/riot-meta#267]
I don't know if anything needs to change on the clients to correctly handle these errors, other than possibly ensuring that any custom error message makes sense for invites (rather than just room creation).

See https://github.com/matrix-org/matrix-doc/pull/1866

**Comment by bmarty** [comment]
Hi @erikjohnston do you think we can get an invite for such a room to do some test?
CC @manuroe

**Comment by erikjohnston** [comment]
This is when sending the invite, rather than receiving the invite. (Also, the synapse PR hasn't yet landed https://github.com/matrix-org/synapse/pull/4721 )

**Comment by lampholder** [comment]
Right, so if I invite @ bob:oldserver.com, and oldserver.com doesn't support your fancy new room version, then this error will be in the synchronous response I get back from this request.
